client/client.go

```diff
@@ -837,6 +834,29 @@ func (r *NotaryRepository) bootstrapRepo() error {
 	return nil
 }
 
+// initializeFromCache looks for cached metadata to bootstrap from
```

Comment:
Alternately, in the `publish` logic, if `ErrMetadataNotFound` is returned by `bootstrapRepo`, we can then call `Initialize` rather than return the previous `ErrRepoNotInitialized`.